Libraries/Promise.js

```diff
@@ -11,13 +11,17 @@
  */
 'use strict';
 
+const prettyFormat = require('pretty-format');
 const Promise = require('fbjs/lib/Promise.native');
 
 if (__DEV__) {
   require('promise/setimmediate/rejection-tracking').enable({
     allRejections: true,
     onUnhandled: (id, error = {}) => {
-      const {message = null, stack = null} = error;
+      let {message = null, stack = null} = error;
```

Comment:
It'd be useful to handle `null` errors and include the error type (TypeError, ReferenceError, etc) too while we're at it. Can you change this method to this implementation?

```
onUnhandled: (id, error: any) => {
  let message: string;
  let stack: ?string;
  
  const stringValue = Object.prototype.toString.call(error);
  if (stringValue === '[object Error]') {
    message = Error.prototype.toString.call(error);
    stack = error.stack;
  } else {
    message = prettyFormat(error);
  }
  
  const warning =
    `Possible Unhandled Promise Rejection (id: ${id}):\n` +
    `${message}\n` +
    (stack == null ? '' : stack);
  console.warn(warning);
}
```